device/coordinate_manager.cpp

```diff
@@ -405,6 +405,26 @@ void CoordinateManager::assert_create_coordinate_manager(
     }
 }
 
+void CoordinateManager::shuffle_tensix_harvesting_mask(const std::vector<uint32_t>& harvesting_locations) {
+    std::vector<uint32_t> sorted_harvesting_locations = harvesting_locations;
+    std::sort(sorted_harvesting_locations.begin(), sorted_harvesting_locations.end());
+    size_t new_harvesting_mask = 0;
+    uint32_t pos = 0;
+    while (tensix_harvesting_mask > 0) {
```

Comment:
Consider:
```
for (uint32_t pos = 0; pos < harvesting_locations.size() && tensix_harvesting_mask > 0; pos ++, tensix_harvesting_mask >>= 1)
```

Reply:
I think while is a bit more readable here because we are not going to do some calculation for every `pos` index, it depends a bit more on the harvesting mask. Not strong opionions on this however, it is probably equally readable